pkg/hintrunner/utils/math_utils.go

```diff
@@ -23,3 +45,67 @@ func AsIntBig(value *big.Int) *big.Int {
 	}
 	return new(big.Int).Sub(value, fp.Modulus())
 }
+
+func div_mod(n, m, p *big.Int) (*big.Int, error) {
+	// https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/python/math_utils.py#L26
+
+	a, _, c := igcdex(m, p)
+	if c.Cmp(big.NewInt(1)) != 0 {
+		return nil, errors.New("no solution exists (gcd(m, p) != 1)")
+	}
+	res := new(big.Int)
+	res.Mul(n, a)
+	res.Mod(res, p)
+	return res, nil
+}
+
+func igcdex(a, b *big.Int) (*big.Int, *big.Int, *big.Int) {
```

Comment:
Return a `big.Int` and not a pointer to it. I've seen this on several occasions on fucntions down the line. Let's start working towards this other pattern PR by PR.